component/core/src/main/java/org/exoplatform/social/core/jpa/storage/RDBMSSpaceStorageImpl.java

```diff
@@ -786,4 +786,24 @@ public void setIdentityStorage(IdentityStorage identityStorage) {
   public int countExternalMembers(Long spaceId) {
     return spaceMemberDAO.countExternalMembers(spaceId);
   }
+
+  public List<Space> getCommonSpaces(String userId, String profileUserId,int offset, int limit)  {
+
+    int countCommonSpaces = spaceDAO.countCommonSpaces(userId, profileUserId);
+    if (countCommonSpaces == 0) {
+      return Collections.emptyList();
+    }
+    List<Space> spaces = new ArrayList<>();
+
+    List<SpaceEntity> commonSpaces = spaceDAO.getCommonSpaces(userId, profileUserId, offset, limit);
+
+    for(SpaceEntity sp : commonSpaces){
+      spaces.add(fillSpaceFromEntity(sp));
+    }
+    return spaces;
```

Comment:
can you use a stream to do this loop ? 
In addition, this will allow you to remove the variable spaces